src/_guides/language/effective-dart/design.md

```diff
@@ -1281,8 +1281,10 @@ num highScore(List<num> scores) {
 }
 {% endprettify %}
 
+<!-- Q: As of 2.9, this code no longer works.
```

Comment:
@munificent opinion?

Reply:
Why doesn't it work? Because we're getting rid of implicit downcasts?

Reply:
> Why doesn't it work? Because we're getting rid of implicit downcasts?

Yes, I think so. Here's the message, which complains about `highest` (an `int`) being assigned `score` (a `num`). 

error * A value of type 'num' can't be assigned to a variable of type 'int'. * lib/effective_dart/design_bad.dart:79:38 * invalid_assignment

```
  num highScore(List<num> scores) {
    var highest = 0;
    for (var score in scores) {
      // ignore: invalid_assignment
      if (score > highest) highest = score;
    }
    return highest;
  }
```